src/pykeen/losses.py

```diff
@@ -510,6 +512,144 @@ def forward(
+            A margin activation. Defaults to ``'relu'``, i.e. $h(\Delta) = max(0, \Delta + \lambda)$, which is the
+            default "margin loss". Using ``'softplus'`` leads to a "soft-margin" formulation as discussed in
+            https://arxiv.org/abs/1703.07737.
+        :param reduction:
+            The name of the reduction operation to aggregate the individual loss values from a batch to a scalar loss
+            value. From {'mean', 'sum'}.
+        """
+        super().__init__(reduction=reduction)
+        if positive_negative_balance <= 0 or positive_negative_balance >= 1:
+            raise ValueError(f"The positive-negative balance weight must be in (0, 1), but is {positive_negative_balance}")
+        if positive_margin < negative_margin:
+            logger.warning("Positive margin should not be smaller than negative margin. Swapping them.")
+            positive_margin, negative_margin = negative_margin, positive_margin
+        self.positive_margin = positive_margin
+        self.negative_margin = negative_margin
+        self.negative_weight = 1.0 - positive_negative_balance
+        self.positive_weight = positive_negative_balance
+        self.margin_activation = margin_activation_resolver.make(margin_activation)
+
+    def process_slcwa_scores(
+        self,
+        positive_scores: torch.FloatTensor,
+        negative_scores: torch.FloatTensor,
+        label_smoothing: Optional[float] = None,
+        batch_filter: Optional[torch.BoolTensor] = None,
+        num_entities: Optional[int] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # Sanity check
+        if label_smoothing:
+            raise UnsupportedLabelSmoothingError(self)
+
+        # positive term
+        if batch_filter is None:
+            # implicitly repeat positive scores
+            loss = self.margin_activation(self.positive_margin - positive_scores) * negative_scores.shape[1]
+        else:
+            # negative_scores have already been filtered in the sampler!
+            num_neg_per_pos = batch_filter.shape[1]
+            positive_scores = positive_scores.unsqueeze(dim=1).repeat(1, num_neg_per_pos, 1)[batch_filter]
+            # shape: (nnz,)
+            loss = self.margin_activation(self.positive_margin - positive_scores)
+        loss = self.positive_weight * self._reduction_method(loss)
+
+        # negative term
+        return loss + self.negative_weight * self._reduction_method(self.negative_margin + negative_scores)
+
+    def process_lcwa_scores(
+        self,
+        predictions: torch.FloatTensor,
+        labels: torch.FloatTensor,
+        label_smoothing: Optional[float] = None,
+        num_entities: Optional[int] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # Sanity check
+        if label_smoothing:
+            labels = apply_label_smoothing(
+                labels=labels,
+                epsilon=label_smoothing,
+                num_classes=num_entities,
+            )
```

Comment:
This is not really possible in the original formulation, since labels are not probability scores. Still I though this would be natural extension.

Reply:
extend away!